drivebrain_app/DriveBrainApp.cpp

```diff
@@ -0,0 +1,239 @@
+    }
+}
+
+DriveBrainApp::DriveBrainApp(int argc, char* argv[])
+    : param_path_(getParamPathFromArgs(argc, argv))
+    , logger_(core::LogLevel::INFO)
+    , config_(param_path_)
+{
+    parseCommandLine(argc, argv);
+    setupSignalHandler();
+    spdlog::set_level(spdlog::level::warn);
+}
+
+DriveBrainApp::~DriveBrainApp() {
+    stop();
+}
+
+void DriveBrainApp::setupSignalHandler() {
+    std::signal(SIGINT, [](int signal) {
+        spdlog::info("Interrupt signal ({}) received. Cleaning up...", signal);
+        DriveBrainApp::stop_signal_.store(true);
+    });
+}
+
+bool DriveBrainApp::initialize() {
+    mcap_logger_ = std::make_unique<common::MCAPProtobufLogger>("temp");
+    
+    controller_ = std::make_unique<control::SimpleController>(logger_, config_);
+    configurable_components_.push_back(controller_.get());
+    
+    matlab_math_ = std::make_unique<estimation::Tire_Model_Codegen_MatlabModel>(
+        logger_, config_, construction_failed_);
+    
+    if (construction_failed_) {
+        stop_signal_.store(true);
+        return false;
+    }
+    
+    configurable_components_.push_back(matlab_math_.get());
+    
+    // live foxglove server server that relies upon having pointers to configurable components for 
+    // getting and handling updates of the registered live parameters
+    foxglove_server_ = std::make_unique<core::FoxgloveWSServer>(configurable_components_);
+    
+    message_logger_ = std::make_shared<core::MsgLogger<std::shared_ptr<google::protobuf::Message>>>(
+        ".mcap", true,
+        std::bind(&common::MCAPProtobufLogger::log_msg, std::ref(*mcap_logger_), std::placeholders::_1),
+        std::bind(&common::MCAPProtobufLogger::close_current_mcap, std::ref(*mcap_logger_)),
+        std::bind(&common::MCAPProtobufLogger::open_new_mcap, std::ref(*mcap_logger_), std::placeholders::_1),
+        std::bind(&core::FoxgloveWSServer::send_live_telem_msg, std::ref(*foxglove_server_), std::placeholders::_1));
+    
+    state_estimator_ = std::make_unique<core::StateEstimator>(logger_, message_logger_, *matlab_math_);
+    
+    driver_ = std::make_unique<comms::CANDriver>(config_, logger_, message_logger_, tx_queue_, io_context_, dbc_path_, construction_failed_, *state_estimator_);
+    
+    if (construction_failed_) {
+        stop_signal_.store(true);
+        return false;
+    }
+    
+    configurable_components_.push_back(driver_.get());
+    
+    eth_driver_ = std::make_unique<comms::MCUETHComms>(logger_, eth_tx_queue_, message_logger_, *state_estimator_,io_context_, "192.168.1.30", 2001, 2000);
+    
+    vn_driver_ = std::make_unique<comms::VNDriver>(config_, logger_, message_logger_, *state_estimator_, io_context_);
+    
+    db_service_ = std::make_unique<DBInterfaceImpl>(message_logger_);
+    
+    // required init, maybe want to call this in the constructor instead
+    bool successful_controller_init = controller_->init();
+
+    return true;
+}
```

Comment:
for all construction of member variables, prefer usage of the constructor method for the `DriveBrainApp` class instead of making an initialization function that is required to be called and if anything fails to construct, throw an exception from within the constructor. 

this will allow us to better follow the RAII pattern:
https://stackoverflow.com/questions/2321511/what-is-meant-by-resource-acquisition-is-initialization-raii